tests/test_preprints.py

```diff
@@ -818,46 +822,114 @@ def user(self):
+                        creator=user,
+                        target=target,
+                        request_type=RequestTypes.WITHDRAWAL.value,
+                        machine_state=DefaultStates.INITIAL.value)
+            request.run_submit(user)
+            return request
+        return withdrawal_request
+
+    @pytest.fixture()
+    def crossref_client(self):
+        return crossref.CrossRefClient(base_url='http://test.osf.crossref.test')
+
+
+    def test_withdrawn_preprint(self, user, preprint, unpublished_preprint_pre_mod, unpublished_preprint_post_mod):
         # test_ever_public
 
         # non-moderated
         assert preprint.ever_public
 
         # pre-mod
-        preprint_pre_mod.run_submit(user)
+        unpublished_preprint_pre_mod.run_submit(user)
 
-        assert not preprint_pre_mod.ever_public
-        preprint_pre_mod.run_reject(user, 'it')
-        preprint_pre_mod.reload()
-        assert not preprint_pre_mod.ever_public
-        preprint_pre_mod.run_accept(user, 'it')
-        preprint_pre_mod.reload()
-        assert preprint_pre_mod.ever_public
+        assert not unpublished_preprint_pre_mod.ever_public
+        unpublished_preprint_pre_mod.run_reject(user, 'it')
+        unpublished_preprint_pre_mod.reload()
+        assert not unpublished_preprint_pre_mod.ever_public
+        unpublished_preprint_pre_mod.run_accept(user, 'it')
+        unpublished_preprint_pre_mod.reload()
+        assert unpublished_preprint_pre_mod.ever_public
 
         # post-mod
-        preprint_post_mod.run_submit(user)
-        assert preprint_post_mod.ever_public
+        unpublished_preprint_post_mod.run_submit(user)
+        assert unpublished_preprint_post_mod.ever_public
 
         # test_cannot_set_ever_public_to_False
-        preprint_pre_mod.ever_public = False
-        preprint_post_mod.ever_public = False
+        unpublished_preprint_pre_mod.ever_public = False
+        unpublished_preprint_post_mod.ever_public = False
         preprint.ever_public = False
         with pytest.raises(ValidationError):
             preprint.save()
         with pytest.raises(ValidationError):
-            preprint_pre_mod.save()
+            unpublished_preprint_pre_mod.save()
         with pytest.raises(ValidationError):
-            preprint_post_mod.save()
+            unpublished_preprint_post_mod.save()
+
+    def test_crossref_status_is_updated(self, make_withdrawal_request, preprint, preprint_post_mod, preprint_pre_mod, moderator, admin, crossref_client):
+        assert preprint.verified_publishable
+        assert preprint_pre_mod.verified_publishable
+        assert preprint_post_mod.verified_publishable
+
+        withdrawal_request = make_withdrawal_request(preprint)
+        withdrawal_request.run_accept(admin, withdrawal_request.comment)
+
+        assert preprint.is_retracted
+        assert not preprint.verified_publishable
+        assert crossref_client.get_status(preprint) == 'unavailable'
```

Comment:
Can you throw one of these above the request creation as a sanity check that it's not `unavailable` before withdrawal?